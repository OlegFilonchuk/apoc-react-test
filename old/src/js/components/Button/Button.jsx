import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { upperFirst } from 'lodash';
import { getDataTestElementProps, dataTestElementPropTypes } from '../../utils/dataTestElementPropUtils';

import './Button.less';

const noop = () => null;

function Button(props) {
  const { className, isSelected, isLinkView, disabled, id, name, type, children, onClick, title, value } = props;

  const buttonClass = classNames('sc-btn', className, {
    'sc-on': isSelected,
    'sc-disabled': disabled,
    'link-view': isLinkView
  });

  const clickHandler = disabled ? noop : onClick;

  const processDataTestElement = dataTestElementValue =>
    `${dataTestElementValue}${value && typeof value === 'string' ? '-' : ''}${
      value && typeof value === 'string' ? upperFirst(value) : ''
    }`;

  return (
    <button
      id={id}
      name={name}
      type={type}
      disabled={disabled}
      className={buttonClass}
      onClick={clickHandler}
      title={title}
      {...getDataTestElementProps(props, processDataTestElement)}
    >
      {children}
    </button>
  );
}

Button.propTypes = {
  /**
   * Button id
   */
  id: PropTypes.string,

  /**
   * Is button disabled
   */
  disabled: PropTypes.bool,

  /**
   * Button name
   */
  name: PropTypes.string,

  /**
   * Button type
   */
  type: PropTypes.string,

  /**
   * Is button selected
   */
  isSelected: PropTypes.bool,

  /**
   * Is button looks like link
   */
  isLinkView: PropTypes.bool,

  /**
   * Is called when button is clicked
   */
  onClick: PropTypes.func.isRequired,

  /**
   * Inherited classes. For more details check Buttons section in Chameleon.
   * Defaults to 'sc-btn-default'.
   */
  className: PropTypes.string,

  /**
   * Button content
   */
  children: PropTypes.node.isRequired,

  /* eslint-disable react/no-unused-prop-types */
  /**
   * Button value that is used in ButtonGroup component
   */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.object]),
  /* eslint-enable react/no-unused-prop-types */

  title: PropTypes.string,

  ...dataTestElementPropTypes
};

Button.defaultProps = {
  id: '',
  name: '',
  type: 'button',
  disabled: false,
  isSelected: false,
  isLinkView: false,
  value: null,
  className: 'sc-btn-default',
  'data-test-element': '',
  title: ''
};

export default Button;
