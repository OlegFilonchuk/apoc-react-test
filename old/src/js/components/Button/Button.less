@import '~chameleon/less/variables.less';

.sc-btn {
  cursor: pointer;

  &.link-view {
    color: @primary;
    background: none;
    padding: initial;

    &.sc-disabled {
      color: @disable;
    }
  }
}
