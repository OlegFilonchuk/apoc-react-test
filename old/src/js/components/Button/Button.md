## Basic button

Simple usage of the **Button** component:

    const callbackFn = () => {
        document.getElementById('clicked').textContent = 'I was clicked!';
    };

    (
      <div>
        <Button className="sc-btn-default" onClick={ callbackFn }>Push Me</Button>
        <div id="clicked">
            Please click button!
        </div>
      </div>
    )

## Disabled button

Rendering Button with a disables state

    <Button onClick={() => true} disabled={true}>You can't click me</Button>

## Colors

Button component can support different colors:

    const onClick = (text) => {
      console.log(`${text} clicked!`);
    };

    (
      <div>
        <p>
          <Button onClick={() => { onClick('Button normal'); }}>Button Normal</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Button Primary'); }}  className="sc-btn-primary">Button Primary</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Primary Outline'); }}  className="sc-btn-primary-outline">Button Primary Outline</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Button Warning'); }}  className="sc-btn-warning">Button Warning</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Button Error'); }}  className="sc-btn-error">Button Error</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Error Outline'); }}  className="sc-btn-error-outline">Button Error Outline</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Button Success'); }}  className="sc-btn-success">
            <i className="sc-icon-download" /> Button Success
          </Button>
        </p>
        <p>
          <Button isLinkView onClick={() => { onClick('Link Button'); }}  className="sc-primary">
            <i className="sc-icon-file" /> Link Button
          </Button>
        </p>
      </div>
    )
