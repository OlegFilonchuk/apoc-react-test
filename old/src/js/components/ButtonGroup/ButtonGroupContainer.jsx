import React from 'react';
import PropTypes from 'prop-types';

import ButtonGroup from '../ButtonGroup/ButtonGroup';
import numberInRange from '../../utils/validation/numberInRange';
/**
 * Button Group component
 */
class ButtonGroupContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedIndex: props.selectedIndex
    };

    this.onChildSelected = this.onChildSelected.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.selectedIndex !== this.state.selectedIndex) {
      this.setState({
        selectedIndex: nextProps.selectedIndex
      });
    }
  }

  onChildSelected(child, index) {
    this.setState({ selectedIndex: index });
    this.onChange(child.props.value);
  }

  onChange(value) {
    this.props.onChange(value);
  }

  get selectedIndex() {
    return this.props.useOnlyPropsSelectedIndex ? this.props.selectedIndex : this.state.selectedIndex;
  }

  render() {
    const childrenWithProps = React.Children.map(this.props.children, (child, index) => {
      const isSelected = index === this.selectedIndex;

      return React.cloneElement(child, {
        isSelected,
        className: 'sc-btn-primary-outline',
        onClick: (...args) => {
          if (child.props.onClick) {
            child.props.onClick(...args);
          }
          this.onChildSelected(child, index, ...args);
        }
      });
    });

    return <ButtonGroup className={this.props.className}>{childrenWithProps}</ButtonGroup>;
  }
}

ButtonGroupContainer.propTypes = {
  /**
   * It will be called when one of the options is clicked
   */
  onChange: PropTypes.func.isRequired,

  /**
   * Children `<Button/>` elements
   */
  children: PropTypes.node.isRequired,

  /**
   * Index of selected Button. By default nothing is selected (-1).
   * Should be between -1 and number of children.
   */
  selectedIndex: function selectedIndex(props, propName, componentName) {
    const min = -1;
    const max = props.children.length - 1;
    const numberInRangeValidation = numberInRange(min, max);

    return numberInRangeValidation(props, propName, componentName);
  },

  /**
   * Component class name
   */
  className: PropTypes.string,

  /**
   * If set to true - only selectedIndex provided by props will be used
   */
  useOnlyPropsSelectedIndex: PropTypes.bool
};

ButtonGroupContainer.defaultProps = {
  selectedIndex: -1,
  className: '',
  useOnlyPropsSelectedIndex: false
};

export default ButtonGroupContainer;
