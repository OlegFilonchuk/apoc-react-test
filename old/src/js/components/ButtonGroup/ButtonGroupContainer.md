## Basic group

    const isDisabled = true;
    const isLicenseMissing = true;
    const value = { id: 'freoifaeorf' };

    function callbackFn(value) {
        const text = JSON.stringify(value)
        document.getElementById('button-group-demo-2').textContent = `You have clicked on option with value "${text}"`;
    }

    function onClick() {
      console.log('Button from ButtonGroupContainer clicked');
    }

    (<div>
        <ButtonGroupContainer onChange={callbackFn} selectedIndex={0}>
          <Button onClick={onClick} value={value}>Option 1</Button>
          <Button onClick={onClick} value="id">Option 2</Button>
          <Button onClick={onClick} disabled={isDisabled}>Option 3</Button>
          <Button onClick={onClick} disabled={isLicenseMissing} isLicenseMissing={true}>Option 4</Button>
        </ButtonGroupContainer>

        <p id="button-group-demo-2">
            Please click button!
        </p>
    </div>)

## Change selected button from the outside

    initialState = {
      index: 0
    }

    function log(option) {
        setState({ index: +option.value })

    }

    function isSelected(i) {
      return state.index === i;
    }

    function changeSelected(e) {
      if (e.target.value) {
        setState({ index: +e.target.value })
      }
    }

    const fakeHandler = () => true;

    (<div>
        <label>Change selected option index for the dropdown below</label>
        <select style={{display:'block', marginBottom:'10px'}} value={state.index} onChange={changeSelected} >
          <option value="0">Option 0</option>
          <option value="1">Option 1</option>
          <option value="2">Option 2</option>
          <option value="3">Option 3</option>
        </select>
        <ButtonGroupContainer
          selectedIndex={state.index}
          onChange={fakeHandler}
        >
          <Button onClick={fakeHandler} value="someValue">Option 0</Button>
          <Button onClick={fakeHandler} value="id">Option 1</Button>
          <Button onClick={fakeHandler}>Option 2</Button>
          <Button onClick={fakeHandler}>Option 3</Button>
        </ButtonGroupContainer>
    </div>)
