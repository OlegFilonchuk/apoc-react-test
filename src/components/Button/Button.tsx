import React from 'react';

export type ButtonProps = {
  label: string;
};

export function Button(props: ButtonProps) {
  return <button>{props.label}</button>;
}
