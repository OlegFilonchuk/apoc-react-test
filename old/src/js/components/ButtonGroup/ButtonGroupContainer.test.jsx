import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import ButtonGroupContainer from './ButtonGroupContainer';
import Button from '../Button/Button';

describe('<ButtonGroupContainer />', () => {
  let onGroupButtonClick;
  const buttonValue = { id: 'test' };
  const onClick = () => true;

  const defaultButtons = [
    <Button onClick={onClick} key={0} value={buttonValue}>
      Option 1
    </Button>,
    <Button onClick={onClick} key={1} value="id">
      Option 2
    </Button>,
    <Button onClick={onClick} key={2} disabled>
      Option 3
    </Button>,
    <Button onClick={onClick} key={3} disabled>
      Option 4
    </Button>
  ];

  beforeEach(() => {
    onGroupButtonClick = jest.fn();
  });

  const mountButtonGroupContainer = (props = {}, buttons = defaultButtons) =>
    mount(
      <ButtonGroupContainer onChange={onGroupButtonClick} {...props}>
        {buttons}
      </ButtonGroupContainer>
    );

  it('should render ButtonGroupContainer component with no Button selected by default', () => {
    const buttonGroupContainer = mountButtonGroupContainer();

    const selectedIndex = buttonGroupContainer.state('selectedIndex');

    expect(selectedIndex).toEqual(-1);

    expect(toJson(buttonGroupContainer)).toMatchSnapshot();
  });

  it('should render ButtonGroupContainer component with 1st Button selected', () => {
    const buttonGroupContainer = mountButtonGroupContainer({ selectedIndex: 0 });

    const selectedIndex = buttonGroupContainer.state('selectedIndex');

    expect(selectedIndex).toEqual(0);

    expect(toJson(buttonGroupContainer)).toMatchSnapshot();
  });

  it('should change selected Button when changed through props', () => {
    const buttonGroupContainer = mountButtonGroupContainer({ selectedIndex: 0 });

    buttonGroupContainer.setProps({ selectedIndex: 1 });

    const selectedIndex = buttonGroupContainer.state('selectedIndex');

    expect(selectedIndex).toEqual(1);

    expect(toJson(buttonGroupContainer)).toMatchSnapshot();
  });

  it('should change selected Button when clicked', () => {
    const buttonGroupContainer = mountButtonGroupContainer({ selectedIndex: 0 });

    buttonGroupContainer
      .find(Button)
      .at(1)
      .simulate('click');

    const selectedIndex = buttonGroupContainer.state('selectedIndex');

    expect(selectedIndex).toEqual(1);
    expect(toJson(buttonGroupContainer)).toMatchSnapshot();
  });

  it('should call onChange callback on button click', () => {
    const buttonGroupContainer = mountButtonGroupContainer();

    buttonGroupContainer
      .find(Button)
      .at(0)
      .simulate('click');

    expect(onGroupButtonClick).toBeCalled();
  });

  it('should pass button value to onChange callback', () => {
    const buttonGroupContainer = mountButtonGroupContainer();

    buttonGroupContainer
      .find(Button)
      .at(0)
      .simulate('click');

    expect(onGroupButtonClick).toBeCalledWith(buttonValue);
  });
});
