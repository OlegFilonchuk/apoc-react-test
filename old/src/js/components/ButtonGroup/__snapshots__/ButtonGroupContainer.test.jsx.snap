// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroupContainer /> should change selected Button when changed through props 1`] = `
<ButtonGroupContainer
  className=""
  onChange={[Function]}
  selectedIndex={1}
  useOnlyPropsSelectedIndex={false}
>
  <ButtonGroup
    className=""
  >
    <div
      className="panel-group"
    >
      <div
        className="sc-btn-switch"
      >
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={false}
          id=""
          isLinkView={false}
          isSelected={false}
          key=".$0"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value={
            Object {
              "id": "test",
            }
          }
        >
          <button
            className="sc-btn sc-btn-primary-outline"
            data-test-element="undefined"
            disabled={false}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 1
          </button>
        </Button>
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={false}
          id=""
          isLinkView={false}
          isSelected={true}
          key=".$1"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value="id"
        >
          <button
            className="sc-btn sc-btn-primary-outline sc-on"
            data-test-element="undefined-Id"
            disabled={false}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 2
          </button>
        </Button>
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={true}
          id=""
          isLinkView={false}
          isSelected={false}
          key=".$2"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value={null}
        >
          <button
            className="sc-btn sc-btn-primary-outline sc-disabled"
            data-test-element="undefined"
            disabled={true}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 3
          </button>
        </Button>
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={true}
          id=""
          isLinkView={false}
          isSelected={false}
          key=".$3"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value={null}
        >
          <button
            className="sc-btn sc-btn-primary-outline sc-disabled"
            data-test-element="undefined"
            disabled={true}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 4
          </button>
        </Button>
      </div>
    </div>
  </ButtonGroup>
</ButtonGroupContainer>
`;

exports[`<ButtonGroupContainer /> should change selected Button when clicked 1`] = `
<ButtonGroupContainer
  className=""
  onChange={[Function]}
  selectedIndex={0}
  useOnlyPropsSelectedIndex={false}
>
  <ButtonGroup
    className=""
  >
    <div
      className="panel-group"
    >
      <div
        className="sc-btn-switch"
      >
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={false}
          id=""
          isLinkView={false}
          isSelected={false}
          key=".$0"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value={
            Object {
              "id": "test",
            }
          }
        >
          <button
            className="sc-btn sc-btn-primary-outline"
            data-test-element="undefined"
            disabled={false}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 1
          </button>
        </Button>
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={false}
          id=""
          isLinkView={false}
          isSelected={true}
          key=".$1"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value="id"
        >
          <button
            className="sc-btn sc-btn-primary-outline sc-on"
            data-test-element="undefined-Id"
            disabled={false}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 2
          </button>
        </Button>
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={true}
          id=""
          isLinkView={false}
          isSelected={false}
          key=".$2"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value={null}
        >
          <button
            className="sc-btn sc-btn-primary-outline sc-disabled"
            data-test-element="undefined"
            disabled={true}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 3
          </button>
        </Button>
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={true}
          id=""
          isLinkView={false}
          isSelected={false}
          key=".$3"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value={null}
        >
          <button
            className="sc-btn sc-btn-primary-outline sc-disabled"
            data-test-element="undefined"
            disabled={true}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 4
          </button>
        </Button>
      </div>
    </div>
  </ButtonGroup>
</ButtonGroupContainer>
`;

exports[`<ButtonGroupContainer /> should render ButtonGroupContainer component with 1st Button selected 1`] = `
<ButtonGroupContainer
  className=""
  onChange={[Function]}
  selectedIndex={0}
  useOnlyPropsSelectedIndex={false}
>
  <ButtonGroup
    className=""
  >
    <div
      className="panel-group"
    >
      <div
        className="sc-btn-switch"
      >
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={false}
          id=""
          isLinkView={false}
          isSelected={true}
          key=".$0"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value={
            Object {
              "id": "test",
            }
          }
        >
          <button
            className="sc-btn sc-btn-primary-outline sc-on"
            data-test-element="undefined"
            disabled={false}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 1
          </button>
        </Button>
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={false}
          id=""
          isLinkView={false}
          isSelected={false}
          key=".$1"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value="id"
        >
          <button
            className="sc-btn sc-btn-primary-outline"
            data-test-element="undefined-Id"
            disabled={false}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 2
          </button>
        </Button>
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={true}
          id=""
          isLinkView={false}
          isSelected={false}
          key=".$2"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value={null}
        >
          <button
            className="sc-btn sc-btn-primary-outline sc-disabled"
            data-test-element="undefined"
            disabled={true}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 3
          </button>
        </Button>
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={true}
          id=""
          isLinkView={false}
          isSelected={false}
          key=".$3"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value={null}
        >
          <button
            className="sc-btn sc-btn-primary-outline sc-disabled"
            data-test-element="undefined"
            disabled={true}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 4
          </button>
        </Button>
      </div>
    </div>
  </ButtonGroup>
</ButtonGroupContainer>
`;

exports[`<ButtonGroupContainer /> should render ButtonGroupContainer component with no Button selected by default 1`] = `
<ButtonGroupContainer
  className=""
  onChange={[Function]}
  selectedIndex={-1}
  useOnlyPropsSelectedIndex={false}
>
  <ButtonGroup
    className=""
  >
    <div
      className="panel-group"
    >
      <div
        className="sc-btn-switch"
      >
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={false}
          id=""
          isLinkView={false}
          isSelected={false}
          key=".$0"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value={
            Object {
              "id": "test",
            }
          }
        >
          <button
            className="sc-btn sc-btn-primary-outline"
            data-test-element="undefined"
            disabled={false}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 1
          </button>
        </Button>
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={false}
          id=""
          isLinkView={false}
          isSelected={false}
          key=".$1"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value="id"
        >
          <button
            className="sc-btn sc-btn-primary-outline"
            data-test-element="undefined-Id"
            disabled={false}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 2
          </button>
        </Button>
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={true}
          id=""
          isLinkView={false}
          isSelected={false}
          key=".$2"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value={null}
        >
          <button
            className="sc-btn sc-btn-primary-outline sc-disabled"
            data-test-element="undefined"
            disabled={true}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 3
          </button>
        </Button>
        <Button
          className="sc-btn-primary-outline"
          data-test-element=""
          disabled={true}
          id=""
          isLinkView={false}
          isSelected={false}
          key=".$3"
          name=""
          onClick={[Function]}
          title=""
          type="button"
          value={null}
        >
          <button
            className="sc-btn sc-btn-primary-outline sc-disabled"
            data-test-element="undefined"
            disabled={true}
            id=""
            name=""
            onClick={[Function]}
            title=""
            type="button"
          >
            Option 4
          </button>
        </Button>
      </div>
    </div>
  </ButtonGroup>
</ButtonGroupContainer>
`;
